app/main.py

```diff
@@ -1,2 +1,71 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    capacity = 8
+    load_factor = 2 / 3
+    size = 0
+
+    def __init__(self) -> None:
+        self.hash_table = self.create_hash_table(8)
+
+    def __len__(self) -> int:
+        return self.size
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        self.size += 1
+        if self.size > self.load_factor * self.capacity:
+            self.resize()
+        index = hash(key) % self.capacity
+
+        keys = [key[0] for key in self.hash_table]
+        if key in keys:
+            while keys[index] != key:
+                index += 1
+                if index > self.capacity:
+                    index = 0
+            self.hash_table[index] = [key, value]
+            self.size -= 1
+        else:
+            while self.hash_table[index] != [None, None]:
+                index += 1
+                if index > self.capacity:
+                    index = 0
+            self.hash_table[index] = [key, value]
```

Comment:
Also look , your logic is repeating in the different methods ( same while loop , almost same cheks etc ) . Try to create some func to reuse your logic